### ISSUE TITLE
[TASK] set_tx_power support 

### ISSUE THREAD

**Issue by tomereli** [prplfoundation/prplMesh#983]
## Description

Add support for setting tx power limit in the agent per radio. This is not available in hostapd, so we need to use nl80211 for that.

### Steps

- [x] Add a new method to nl80211_client to set tx power limit using NL80211_CMD_SET_WIPHY.
- [x] Add a nl80211_client to the base_wlan_hal which will be used by all BWL backends to get and set WLAN nl80211 generic commands.
- [x] Move `set_tx_power_limit` API to ap_wlan_hal
- [x] Implement the stub in `set_tx_power_limit` to use the `nl80211_client` for setting tx_power limit
- [ ] Implement set_tx_power limit in dummy BWL implementation which will store the limit
- [ ] Enhance channel selection dummy test to set new tx power limit as done in the certification

Edit - the last two are a bit tricky with the current implementation since the `nl80211_client` doesn't have access to the `m_radio_info` which holds the tx power reading - this is read from hostapd using `radio_info` command which reports the tx power via the TxPower parameter, and is saved in the `m_radio_info` structure which belongs to the `base_wlan_hal` class.
So, unless there are better ideas I suggest not to implement this in dummy mode at all. We will have coverage for this in real devices test flows.

**Comment by vitalybu** [comment]
`set_tx_power_limit` should probably remain in `ap_wlan_hal`.
I don't think that the monitor will ever need to set the power limit (nor it should be able to).

**Comment by tomereli** [comment]
> `set_tx_power_limit` should probably remain in `ap_wlan_hal`.
> I don't think that the monitor will ever need to set the power limit (nor it should be able to).

Fixed (and already implemented like that - see #982)

**Comment by tomereli** [comment]
So apparently setting tx power is simply not supported in the driver:
```
root@OpenWrt:/# iw --debug dev wlan2 set txpower limit 500
-- Debug: Sent Message:
--------------------------   BEGIN NETLINK MESSAGE ---------------------------
  [NETLINK HEADER] 16 octets
    .nlmsg_len = 44
    .type = 22 <0x16>
    .flags = 5 <REQUEST,ACK>
    .seq = 1579344540
    .port = -1832895688
  [GENERIC NETLINK HEADER] 4 octets
    .cmd = 2
    .version = 0
    .unused = 0
  [PAYLOAD] 24 octets
    00 08 00 03 00 00 00 0a 00 08 00 61 00 00 00 01 ...........a....
    00 08 00 62 00 00 01 f4                         ...b....
---------------------------  END NETLINK MESSAGE   ---------------------------
-- Debug: Received Message:
--------------------------   BEGIN NETLINK MESSAGE ---------------------------
  [NETLINK HEADER] 16 octets
    .nlmsg_len = 64
    .type = 2 <ERROR>
    .flags = 0 <>
    .seq = 1579344540
    .port = -1832895688
  [ERRORMSG] 20 octets
    .error = -122 "Not supported"
  [ORIGINAL MESSAGE] 16 octets
    .nlmsg_len = 16
    .type = 22 <0x16>
    .flags = 5 <REQUEST,ACK>
    .seq = 1579344540
    .port = -1832895688
---------------------------  END NETLINK MESSAGE   ---------------------------
command failed: Not supported (-122)
root@OpenWrt:/#
```

And here the same command working on UGW with the latest driver:
```
root@ugwcpe:/# iw dev wlan2 set txpower limit 500
[26446.009046] [0002614599] mtlk0(_wv_ieee80211_op_config:1568): phy1: Power level change, power_level=5, max_power=30, max_reg_power=30
[26446.031420] [0002614601] mtlk0(_mtlk_coc_set_antennas:346): CID-0107: CoC power mode changed from TX2xRX2 to TX2xRX2
```

**This means this task depends on #889**
